partcad/tests/unit/test_git_clone_retries.py

```
@@ -0,0 +1,119 @@
+def test_project_import_git_clone_retry_failure(git_error: GitCommandError):
+    def side_effect(*args, **kwargs):
+        side_effect.counter += 1
+        raise git_error
+    side_effect.counter = 0
+
+    ctx = pc.Context()
+    factory = pc.ProjectFactoryGit(ctx, None, test_config_import_git)
+    with patch("git.Repo.clone_from", side_effect=side_effect) as mock_clone_from, \
+         patch.object(pc.user_config, "git_retry_config", test_git_retry_config), \
+         patch("builtins.open", mock_open(read_data="")):
+        with pytest.raises(RuntimeError):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)